src/index.js

```diff
@@ -9,27 +9,39 @@ const { Template } = require("./messages/template");
 const Text = require("./messages/text");
 
 const api = require("./fetch");
+const { post, get } = require("./requests");
+
+const EventEmitter = require("node:events")
+const { createHmac } = require('node:crypto');
 
 /**
  * The main API object
  *
  * @property {String} token The API token
+ * @property {String} appSecret The app secret
+ * @property {String} webhookVerifyToken The webhook verify token
  * @property {String} v The API version to use
  * @property {Boolean} parsed If truthy, API operations will return the fetch promise instead. Intended for low level debugging.
  */
-class WhatsAppAPI {
+class WhatsAppAPI extends EventEmitter {
     /**
      * Initiate the Whatsapp API app
      *
      * @param {String} token The API token, given at setup. It can be either a temporal token or a permanent one.
+     * @param {String} appSecret The app secret, given at setup.
+     * @param {String} webhookVerifyToken The webhook verify token, given at setup.
```

Comment:
Minor wording change, rather than `given at setup` it could be `configured at setup`, since the user is the one that creates the verify token and not the API.

Reply:
Yes, I agree